### PR TITLE
Add no-empty-first-line rule

### LINKED ISSUE #3536: Add no-empty-first-line
> What is the problem you're trying to solve?

I would like to enforce a rule similar to `no-missing-end-of-source-newline` but preventing to start the css file from the empty line.

Example of incorrect files:

```

@import "..."
...
```

```

.bar {
   ...
}
```

I could not find a setup which would allow me to prevent the first line of the file to be empty using currently available rules.

### PR DESCRIPTION
**Pull request by Bilie**
Closes #3536

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/no-empty-first-line/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Disallow empty first lines.
+
+```css
+    \n
+    /** ↑
+     * This newline */
+    a { color: pink; }
+```
+
+Completely empty files are not considered violations. These are handled by the rule [no-empty-source](../no-empty-source/README.md).
```
Comment (jeddy3) [inline lib/rules/no-empty-first-line/README.md]: I think the following is more in line with the language in other rule READMEs (and sources aren't always files):

```
This rule ignores empty sources. Use the [`no-empty-source`](../no-empty-source/README.md) rule to disallow these.
```

File: lib/rules/no-empty-first-line/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+const { messages, ruleName } = rule;
+
+testRule(rule, {
+  ruleName,
+  config: [true],
+  skipBasicChecks: true,
+  fix: true,
+
+  accept: [
+    {
```
Comment (jeddy3) [inline lib/rules/no-empty-first-line/__tests__/index.js]: Can we get tests for single-line and multiline whitespace, please? e.g.

```js
  {
    code: "  "
  },
  {
    code: "\n"
  },
  {
    code: "\n\n"
  },
```

To test [this line](https://github.com/stylelint/stylelint/pull/3650/files#diff-0465d1dd14c45b6f34815530242119e8R26).

**Comment by Bilie** [comment]
Thank you very much for reviewing my PR! Enjoy your vacation!! :)

**Comment by jeddy3** [comment]
> -   Added: `no-empty-first-line` rule ([#3650](https://github.com/stylelint/stylelint/pull/3650)).

**Pull request merged**